src/pyfmi/fmi3.pyx

```diff
@@ -542,26 +563,651 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
         if status != 0:
             raise FMUException('Failed to set the Float32 values. See the log for possibly more information.')
 
+    cpdef set_int64(self, valueref, values):
+        """
+        Sets the int64-values in the FMU as defined by the value reference(s).
+
+        Parameters::
+
+            valueref --
+                A list of value references.
+
+            values --
+                Values to be set.
+
+        Example::
+
+            model.set_int64([234, 235],[2, 10])
+
+        Calls the low-level FMI function: fmi3SetInt64
+        """
+        cdef int status
+        cdef np.ndarray[FMIL3.fmi3_value_reference_t, ndim=1, mode='c'] input_valueref = np.asarray(valueref, dtype = np.uint32).ravel()
+        cdef np.ndarray[FMIL3.fmi3_int64_t, ndim=1, mode='c'] set_value = np.asarray(values, dtype = np.int64).ravel()
+
+        if np.size(input_valueref) != np.size(set_value):
+            raise FMUException('The length of valueref and values are inconsistent. Note: Array variables are not yet supported')
```

Comment:
I think we should split this error check into a separate function since we have it now in so many places, and maybe we can also improve the error message to also include the size differences.

Reply:
Split this out now, see changed.

As for the size check, I suggest we wait until we start on arrays and how exactly the inputs will look like then.
I'd imagine these would end up something like `fmu.set([1, 2], [[1, 2, 3], [[1, 2], [3, 4]]])` + then one could check against the actual sizes of the referenced variables.

Reply:
Also, I had considered if we could do something more along the lines of macros/inlining, but there would appear to be some issues with the class here, see comment under inlining here: https://cython.readthedocs.io/en/latest/src/userguide/pyrex_differences.html#cdef-inline

Reply:
Nice. I agree with your comment (resolve).